atc/db/worker_task_cache.go

```diff
@@ -0,0 +1,113 @@
+package db
+
+import (
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+type WorkerTaskCache struct {
+	WorkerName string
+	TaskCache  UsedTaskCache
+}
+
+type UsedWorkerTaskCache struct {
+	ID         int
+	WorkerName string
+}
+
+func (wtc WorkerTaskCache) FindOrCreate(
+	tx Tx,
+) (*UsedWorkerTaskCache, error) {
+	var id int
+	err := psql.Select("id").
+		From("worker_task_caches").
+		Where(sq.Eq{
+			"worker_name":   wtc.WorkerName,
+			"task_cache_id": wtc.TaskCache.ID(),
+		}).
+		RunWith(tx).
+		QueryRow().
+		Scan(&id)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			err = psql.Insert("task_caches").
```

Comment:
why do we also insert a new task cache here? Would it not already exist since the `UsedTaskCache` exists?

Reply:
When coming from volume.go `InitializeTaskCache`, the task cache is not in DB yet. So it has to create a new task cache first here. 

But I can see the reason that this part of the code doesn't belongs to worker task cache. Maybe, we should init the task cache in `InitializeTaskCache` first?